internal/adapters/dataproviders/database/mongo/pegin.go

```diff
@@ -208,3 +240,96 @@ func (repo *peginMongoRepository) DeleteQuotes(ctx context.Context, quotes []str
 	}
 	return uint(peginResult.DeletedCount + retainedResult.DeletedCount + creationDataResult.DeletedCount), nil
 }
+
+func (repo *peginMongoRepository) GetQuotesByState(ctx context.Context, filter quote.GetPeginQuotesByStateFilter) ([]quote.PeginQuote, error) {
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+
+	quoteHashes, err := repo.getRetainedQuoteHashes(dbCtx, filter.States)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(quoteHashes) == 0 {
+		return []quote.PeginQuote{}, nil
+	}
+
+	peginQuotes, err := repo.getPeginQuotesByHashesAndDate(dbCtx, quoteHashes, filter.StartDate, filter.EndDate)
+	if err != nil {
+		return nil, err
+	}
+
+	logDbInteraction(Read, peginQuotes)
+	return peginQuotes, nil
+}
+
+func (repo *peginMongoRepository) getRetainedQuoteHashes(ctx context.Context, states []quote.PeginState) ([]string, error) {
```

Comment:
you already have a function that does this it returns the array of retained quotes, then you can use the quote hash field without adding the projection